src/ConfigSettings/ConfigSettingsLOQ.tsx

```diff
@@ -7,16 +7,25 @@ import { UploadFile, Edit } from '@mui/icons-material';
 
 // Local data
 import ConfigSettingsGeneral from './ConfigSettingsGeneral';
+import FileUpload from '../FileUpload';
+import { useParams } from 'react-router-dom';
+
+const fiaApiUrl = process.env.REACT_APP_FIA_REST_API_URL;
 
 const ConfigSettingsLOQ: React.FC = () => {
+  const { instrumentName } = useParams<{ instrumentName: string }>();
+  const instrument_url = `${fiaApiUrl}/extras/${instrumentName.toLowerCase()}`;
+  const { response, handleFileUpload } = FileUpload(instrument_url);
   // Insert LOQ specifc buttons into the ConfigSettingsGeneral component
   return (
     <ConfigSettingsGeneral>
-      {/* Upload file button -- disabled for now */}
       <Box sx={{ mb: 2 }}>
-        <Button variant="contained" disabled startIcon={<UploadFile />}>
+        <Button component="label" variant="contained" startIcon={<UploadFile />}>
           Upload file...
+          <input type="file" multiple hidden onChange={(event) => handleFileUpload(event)} />
         </Button>
+        {/* replace with a snackbar/toastlike component */}
+        {response && <pre>{JSON.stringify(response.data, null, 2)}</pre>}
```

Comment:
When a user selects a file to be uploaded, think it would look better so that the file's name appears adjacent (to the right) of the "Upload file" button. And for it NOT to actually fire the API POST request until the user actually hits "Apply settings". 

Would also need to add state changes to `unsavedChanges` and `applyMessage` similar to how I've done it for when the reduction status button is pressed:

```tsx
const toggleEnabledStatus = (): void => {
  setEnabledStatus(!enabledStatus);
  setReductionStatus(enabledStatus ? 'OFF' : 'ON');
  setUnsavedChanges(true);
  setApplyMessage('');
};
```

Reply:
I didn't quite use the existing applyMessage state, and I wasn't sure how to hook into the unsavedStateChanges     
but placed a callback to the fileupload alongside `setApplyMessage('Changes applied successfully'); setUnsavedChanges(false);` to ensure the file only uploads when the apply button is pressed